sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/PeriodicMetricReader.java

```diff
@@ -81,7 +81,21 @@ public MemoryMode getMemoryMode() {
 
   @Override
   public CompletableResultCode forceFlush() {
-    return scheduled.doRun();
+    CompletableResultCode result = new CompletableResultCode();
+    CompletableResultCode flushResult = scheduled.doRun();
+    flushResult.whenComplete(
+        () ->
+            exporter
+                .flush()
+                .whenComplete(
+                    () -> {
+                      if (!flushResult.isSuccess()) {
+                        result.fail();
+                      } else {
+                        result.succeed();
+                      }
+                    }));
+    return result;
```

Comment:
I've seen this pattern in enough places I wonder if it would be worth adding `CompletableResultCode.whenComplete(Callable<CompletableResultCode>)` that handles this logic in one place.

Reply:
I like this idea